# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/self-test/baselineCompile.js
# Commit: 51d604ea785b
# Full Hash: 51d604ea785bbc46db7f88235faef597f57e671f
# Author: Jason Orendorff <jorendorff@mozilla.com>
# Date: 2018-02-24 10:13:47
# Regressor Bug: 1440431
# File Overlap Count: 1
# Description:
#   Bug 1440431 - Part 3: Add baselineCompile() testing function. r=nbp.
# ==============================================================================

diff -r d6d01ad35d89 -r 51d604ea785b js/src/jit-test/tests/self-test/baselineCompile.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/self-test/baselineCompile.js	Thu Feb 22 13:04:46 2018 -0600
@@ -0,0 +1,20 @@
+// Check that the help text for baselineCompile() is accurate.
+
+if (typeof inJit == "function" && typeof baselineCompile == "function") {
+    if (!inJit()) {
+
+        baselineCompile();  // compile the current script
+
+        assertEq(inJit(), false,
+                 "We have compiled this script to baseline jitcode, but shouldn't " +
+                 "be running it yet, according to the help text for baselineCompile() " +
+                 "in TestingFunctions.cpp. If you fail this assertion, nice work, and " +
+                 "please update the help text!");
+
+        for (var i=0; i<1; i++) {}  // exact boilerplate suggested by the help text
+
+        assertEq(inJit(), true,
+                 "help text in TestingFunctions.cpp claims the above loop causes " +
+                 "the interpreter to start running the new baseline jitcode");
+    }
+}
